### PR TITLE
#187742590 User should be able to verify the email

### PR DESCRIPTION
**Pull request by kwizera-bonheur25**
#### What does this PR do?

- User should be able to verify the email

#### Description of Task to be completed?

- User should be able to verify the email

#### How should this be manually tested?

sign up with your email test on the link sent to your email

#### What are the relevant pivotal tracker/Trello stories?

#187742590

### CODE REVIEW

**Review by YvetteNyibuka — CHANGES_REQUESTED** [review]
You still have some issues with responsiveness

image

image

fix them first

File: src/components/Nav.tsx [hunk 1]
```diff
@@ -18,6 +18,12 @@ export default function Nav() {
 					{' '}
 					<Link to="/register">Register</Link>
 				</li>
+				<li>
+					{' '}
+					<Link to="users/account/verify/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImI0NTIzYjdjLTE2YjAtNDQ2Ni05Zjc4LTFlYmU0OTBiZjg2OSIsInJvbGUiOiIxMWFmZDRmMS0wYmVkLTRhM2ItOGFkNS0wOTc4ZGFiZjhmY2QiLCJpYXQiOjE3MTg2NTE4MzgsImV4cCI6MTc1MDIwOTQzOH0.wbLPh_6bzumzUSbqamnvQimke-xuYxIrGAsQoxZ6hpI">
```
Comment (Hakizimana-Clement) [inline src/components/Nav.tsx]: Why do we have this token in this link???????

Comment (Hakizimana-Clement) [inline src/components/Nav.tsx]: Why do we have this token in this link?????

Comment (Hakizimana-Clement) [inline src/components/Nav.tsx]: Why do we have this token in this link???????

Comment (Hakizimana-Clement) [inline src/components/Nav.tsx]: Why do we have this token in this link?????

**Review by Hakizimana-Clement — APPROVED** [review]

**Review by Tuyisenge2 — APPROVED** [review]

File: src/@types/auth/VerifyAccountTypes.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export interface VerifyState {
+	isLoading: boolean;
+	isVerified: boolean;
+	message: string | null;
+}
```
Comment (leandreAlly) [inline src/@types/auth/VerifyAccountTypes.ts]: change this into types instead of interface

File: src/components/auth/VerifyAccount.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import { useParams } from 'react-router-dom';
+import { useAppDispatch } from '../../redux/hooks/hooks';
+import { verify } from '../../redux/features/VerifyAccountSlice';
+import { useEffect } from 'react';
+import useToast from '../../hooks/useToast';
+import { DynamicData } from '../../@types/DynamicData';
+import VerifiedCard from '../cards/VerifiedCard';
+import { useRef } from 'react';
```
Comment (leandreAlly) [inline src/components/auth/VerifyAccount.tsx]: simply organize this import

File: src/components/auth/VerifyAccount.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+const VerifyAccount = () => {
+	const { showErrorMessage, showSuccessMessage } = useToast();
+	const dispatch = useAppDispatch();
+	const { token } = useParams();
+	const effectRan = useRef(false);
+
+	useEffect(() => {
+		if (effectRan.current) return;
+		effectRan.current = true;
```
Comment (leandreAlly) [inline src/components/auth/VerifyAccount.tsx]: remove this useRef hook and check if it's working on hosted version

**Review by leandreAlly — CHANGES_REQUESTED** [review]

**Review by leandreAlly — APPROVED** [review]

**Pull request merged**